File: clients/swift/Sources/Komus24/Model/Quest.swift
/// A delivery quest
public struct Quest {
    /// Cell where to pick delivery from
    let pickupCell: Vec2Int

    /// Cell to drop the delivery at
    let dropCell: Vec2Int

    /// Score for completing the quest
    let score: Int64

    /// Read Quest from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Quest {
        var pickupCell: Vec2Int
        pickupCell = Vec2Int.readFrom(stream)
        var dropCell: Vec2Int
        dropCell = Vec2Int.readFrom(stream)
        var score: Int64
        score = stream.readInt64()
        return Quest(pickupCell: pickupCell, dropCell: dropCell, score: score)
    }

    /// Write Quest to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        pickupCell.writeTo(stream)
        dropCell.writeTo(stream)
        stream.writeInt64(score)
    }
}

File: clients/kotlin_graal/src/main/kotlin/komus24/model/Quest.kt
package komus24.model

import komus24.util.StreamUtil

/**
 * A delivery quest
 */
class Quest {
    /**
     * Cell where to pick delivery from
     */
    var pickupCell: komus24.model.Vec2Int
    /**
     * Cell to drop the delivery at
     */
    var dropCell: komus24.model.Vec2Int
    /**
     * Score for completing the quest
     */
    var score: Long

    constructor(pickupCell: komus24.model.Vec2Int, dropCell: komus24.model.Vec2Int, score: Long) {
        this.pickupCell = pickupCell
        this.dropCell = dropCell
        this.score = score
    }

    /**
     * Write Quest to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        pickupCell.writeTo(stream)
        dropCell.writeTo(stream)
        StreamUtil.writeLong(stream, score)
    }

    /**
     * Get string representation of Quest
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Quest { ")
        stringBuilder.append("pickupCell: ")
        stringBuilder.append(pickupCell)
        stringBuilder.append(", ")
        stringBuilder.append("dropCell: ")
        stringBuilder.append(dropCell)
        stringBuilder.append(", ")
        stringBuilder.append("score: ")
        stringBuilder.append(score)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Quest from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Quest {
            var pickupCell: komus24.model.Vec2Int
            pickupCell = komus24.model.Vec2Int.readFrom(stream)
            var dropCell: komus24.model.Vec2Int
            dropCell = komus24.model.Vec2Int.readFrom(stream)
            var score: Long
            score = StreamUtil.readLong(stream)
            return Quest(pickupCell, dropCell, score)
        }
    }
}